app/post.go

```diff
@@ -369,6 +383,32 @@ func (a *App) CreatePost(c request.CTX, post *model.Post, channel *model.Channel
 	return rpost, nil
 }
 
+func (a *App) validateVoiceMessage(post *model.Post) *model.AppError {
+	fileIds := post.FileIds
+	nbFiles := len(fileIds)
+	if nbFiles == 0 {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.no_file.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+	}
+
+	if nbFiles > 1 {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.too_many_files.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+	}
+
+	fileInfo, err := a.GetFileInfo(fileIds[0])
+	if err != nil {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.unable_get_file_info.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusInternalServerError).Wrap(err)
+	}
+	if fileInfo == nil {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.file_not_found.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusBadRequest)
+	}
+
+	if fileInfo.Size > *a.Config().FileSettings.MaxVoiceMessagesFileSize {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.file_too_big.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusBadRequest)
+	}
```

Comment:
Should we add a check on `Fileinfo.MimeType` as well? We could have a constant list of formats we support (e.g. mp3/m4a).

Reply:
Will add a check! Don't we support mp3 only? :confused:

Reply:
Yes, we can check for mp3 but leave a way to extend it in the future if needed.